pkg/comp-functions/functions/vshnpostgres/pgqexporter_workaround.go

```diff
@@ -0,0 +1,566 @@
+package vshnpostgres
+
+import (
+	"context"
+	"fmt"
+
+	xkubev1 "github.com/crossplane-contrib/provider-kubernetes/apis/object/v1alpha1"
+	"github.com/crossplane/function-sdk-go/proto/v1beta1"
+	vshnv1 "github.com/vshn/appcat/v4/apis/vshn/v1"
+	"github.com/vshn/appcat/v4/pkg/comp-functions/runtime"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+)
+
+func PgExporterWorkaround(ctx context.Context, svc *runtime.ServiceRuntime) *v1beta1.Result {
+	log := controllerruntime.LoggerFrom(ctx)
+
+	comp := &vshnv1.VSHNPostgreSQL{}
+	err := svc.GetObservedComposite(comp)
+	if err != nil {
+		return runtime.NewFatalResult(fmt.Errorf("cannot get composite: %w", err))
+	}
+
+	log.Info("Making sure the cluster exposed connection details")
+	obj := &xkubev1.Object{}
+	err = svc.GetObservedComposedResource(obj, "cluster")
+	if err != nil {
+		return runtime.NewWarningResult(fmt.Sprintf("cannot get the sgcluster object: %s", err))
+	}
+
+	// get configmap
+	configMap := &corev1.ConfigMap{}
+	// create k8s client inside cluster
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		panic(err.Error())
+	}
+	// creates the clientset
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		panic(err.Error())
+	}
+	// get configmap
+	configMap, err = clientset.CoreV1().ConfigMaps(comp.GetNamespace()).Get(context.Background(), comp.GetName()+"-prometheus-postgres-exporter-config", metav1.GetOptions{})
+	if err != nil {
+		return runtime.NewWarningResult(fmt.Sprintf("cannot get configmap: %s", err))
+	}
+
+	//annotate ConfigMap
+	configMap.Annotations["stackgres.io/reconciliation-pause"] = "true"
+	_, err = clientset.CoreV1().ConfigMaps(comp.GetNamespace()).Update(context.Background(), configMap, metav1.UpdateOptions{})
+	if err != nil {
+		return runtime.NewWarningResult(fmt.Sprintf("cannot annotate configmap: %s", err))
+	}
+
+	// replace Data inside ConfigMap
+	configMap.Data["queries.yaml"] = queries
+	_, err = clientset.CoreV1().ConfigMaps(comp.GetNamespace()).Update(context.Background(), configMap, metav1.UpdateOptions{})
+	if err != nil {
+		return runtime.NewWarningResult(fmt.Sprintf("cannot update configmap: %s", err))
+	}
+
+	return nil
+}
+
+const queries = `
```

Comment:
Can you please put this in a file and use go embed? It will make this a million times more manageable.

See an example here: https://github.com/vshn/appcat/blob/master/pkg/comp-functions/functions/vshnmariadb/backup.go#L20